src/ansible_navigator/configuration_subsystem/definitions.py

```diff
@@ -43,43 +46,46 @@ class Constants(Enum):
     USER_CLI = "cli parameters"
 
 
-class SettingsEntryValue(SimpleNamespace):
-    # pylint: disable=too-few-public-methods
-    """An object to store a value"""
+@dataclass
+class SettingsEntryValue:
+    """An object to store a value."""
 
+    #: The default value for the entry
     default: Any = Constants.NOT_SET
```

Comment:
It'd be cool to nuke these `Any`s some day. Perhaps by making `SettingsEntryValue` [polymorphic](https://docs.python.org/3/library/typing.html#generics) over a type parameter that describes what the thing could be? i.e. instead of just a `SettingsEntryValue`, we could have `SettingsEntryValue[T]` and when we use it, it would be `SettingsEntryValue[str]` or whatever the actual type of the setting is.

Internally, we could `Union` it with `Constants` (or some new place we put `Constants.NOT_SET`) so that `NOT_SET` still works.

Just food for future thought.